sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/MetricsAdvisorAdministrationClient/DatasourceCredentialLiveTests.cs

```diff
@@ -12,53 +13,96 @@ namespace Azure.AI.MetricsAdvisor.Tests
     public class DatasourceCredentialLiveTests : MetricsAdvisorLiveTestBase
     {
         private const string ClientId = "clientId";
-        private const string ClientSecret = "clientSecret";
         private const string TenantId = "tenantId";
 
         public DatasourceCredentialLiveTests(bool isAsync) : base(isAsync)
         {
         }
 
         [RecordedTest]
-        public async Task CreateAndGetServicePrincipalDatasourceCredential()
+        [TestCase(nameof(ServicePrincipalDatasourceCredential))]
+        public async Task CreateAndGetDatasourceCredential(string credentialTypeName)
```

Comment:
I'm adding a string parameter so we can test all DatasourceCredential cases without repeating too much code. The string parameter is passed to `GetDatasourceCredentialTestCase` to get the corresponding credential instance.